### PR TITLE
Handle package-specifier spacing

### LINKED ISSUE #195: A PR is not opened for CI - Check dependencies
This is seen now for several repositories - mainly when using the `ignore` input.

### PR DESCRIPTION
**Pull request by CasperWA**
If there is no space between the package name and the version specifiers, the `pyproject.toml` file would not be updated. This has been remedied.

Fixes #195.

In addition, this PR adds a `debug` input value for the _CI - Check dependencies_ callable worfklow, which will utilize the new `--verbose` option for the `update-deps` task.

### CODE REVIEW

File: ci_cd/tasks/update_deps.py [hunk 1]
```diff
@@ -70,10 +72,16 @@ def update_deps(  # pylint: disable=too-many-branches,too-many-locals,too-many-s
...
         ignore_separator: str = ignore_separator  # type: ignore[no-redef]
+        verbose: bool = verbose  # type: ignore[no-redef]
 
     if not ignore:
         ignore: list[str] = []  # type: ignore[no-redef]
 
+    if verbose:
+        LOGGER.setLevel(logging.DEBUG)
+        LOGGER.addHandler(logging.StreamHandler(sys.stdout))
+        LOGGER.debug("Verbose logging enabled.")
```
Comment (ajeklund) [inline ci_cd/tasks/update_deps.py:83]: Codecov reports that these lines are not covered by tests. Could/should they be tested?

File: tests/tasks/test_update_deps.py [hunk 2]
```diff
@@ -149,9 +150,9 @@ def test_update_deps(tmp_path: "Path", caplog: pytest.LogCaptureFixture) -> None
         elif "name1" in line:
             assert line == "name1<=1"
         elif "name2" in line:
```
Comment (ajeklund) [inline tests/tasks/test_update_deps.py:152]: Why did you change these lines?

File: tests/tasks/test_update_deps.py [hunk 3]
```diff
@@ -1158,7 +1159,7 @@ def test_ignore_rules_logic(
 
     for line in dependencies:
         for dependency, dependency_requirement in expected_result.items():
-            if f"{dependency} " in line:
+            if re.match(rf"{re.escape(dependency)}\s*(~|>).*", line):
```
Comment (ajeklund) [inline tests/tasks/test_update_deps.py:1162]: It would be helpful with a comment for understanding this test.

**Review by ajeklund — CHANGES_REQUESTED** [review]
Thank you! I've made a couple of comments.

File: ci_cd/tasks/update_deps.py (hunk 1 above)
Comment (CasperWA) [reply ci_cd/tasks/update_deps.py:83]: No, they are log messages, not part of the actual code as such. As a solution I can either: Run the tests with `--verbose` or add a `# pragma: no cover` to not include this block (these three lines) in the code coverage statistics?

File: tests/tasks/test_update_deps.py (hunk 2 above)
Comment (CasperWA) [reply tests/tasks/test_update_deps.py:152]: This is the best question in this review. I'll try to explain - I don't know what the correct path forward here is, though. Hmm, perhaps I'll explain offline first and add a recap here after the fact?

File: ci_cd/tasks/update_deps.py (hunk 1 above)
Comment (ajeklund) [reply ci_cd/tasks/update_deps.py:83]: If there is a simple way to check that `--verbose` works (i.e. produces additional messages), I would prefer that. If that is complicated or you consider it to be unnecessary, then I accept the the no cover-solution.

Comment (CasperWA) [reply ci_cd/tasks/update_deps.py:83]: It's not complicated - I can add an extra separate test for it :)

File: tests/tasks/test_update_deps.py (hunk 2 above)
Comment (CasperWA) [reply tests/tasks/test_update_deps.py:152]: So - after talking offline, we have agreed that `name3` will be commented out of the test pyproject.toml file with a comment stating why.

File: tests/tasks/test_update_deps.py [hunk 4]
```diff
@@ -149,9 +152,19 @@ def test_update_deps(tmp_path: "Path", caplog: pytest.LogCaptureFixture) -> None
         elif "name1" in line:
             assert line == "name1<=1"
         elif "name2" in line:
-            assert line == "name2>=3"
+            # This is updated to include the latest version (v1.2.3) even though it was
```
Comment (ajeklund) [inline tests/tasks/test_update_deps.py:155]:
```suggestion
            # This is updated to include the latest version (v1.2.3 for all nameX) even though it was
```

**Review by ajeklund — CHANGES_REQUESTED** [review]
One minor suggestion.

File: tests/tasks/test_update_deps.py (hunk 4 above)
Comment (CasperWA) [reply tests/tasks/test_update_deps.py:155]: While this is true, here we're just looking at the `name2` package, which is mocked to have `v1.2.3` be the latest version.
While package `name10` also has the latest version mocked to `v1.2.3` this is not relevant for `name2` and the comment here.

Comment (CasperWA) [reply tests/tasks/test_update_deps.py:155]: discussed offline and updated accordingly :)
